tutorial/storage.md

```diff
@@ -0,0 +1,137 @@
+
+ls ./data/swift/ | xargs -I {} -n 1 bash -c "ralph push -b swift {}"
+```
+
+The `list` command will return the name of all archives stored in Swift backend.
+You can check if all the files have been correctly pushed in the backend.
+
+Run the following command to list all archives stored in Swift backend:
+
+```
+# list all files in Swift backend
+
+$ ralph list -b swift
+```
+
+It should return the following result:
+
+```
+edx_archive.json
+xapi_archive.json
+```
+
+## Convert logs to xAPI format
+
+Converting data in `ralph` requires to use the `fetch`, `extract` and `convert` command.
+
+Now that data are successfully stored in the input backend, you will fetch the edX-named
+archive, extract and convert its content from edX to xAPI format.
+
+### Step-by-step conversion
+
+As the conversion requires to use three commands, we will first go step-by-step.
+
+First of all, run the following command to `fetch` the edX archive:
+
+```
+# fetch edx_archive.json in Swift backend
+
+$ ralph fetch -b swift edx_archive.json
+```
+
+It should return:
+
+```
+<!-- return fetch edx_archive  -->
+```
+
+The data are stored in [GELF format](https://docs.graylog.org/docs/gelf) in the archive
+and we need to `extract` the relevant data from statements with a GELF-customised parser.
+
+To extract data from the edX archive by parsing them, run the following command:
+
+```
+# fetch edX archive and extract its content
+
+$ ralph fetch -b swift edx_archive.json | ralph extract -p gelf
+```
+
+It should return:
+
+```
+<!-- return fetch edx_archive  -->
+```
+
+### Convert from edX to xAPI
+
+The next step is to store the extracted logs in the xAPI format.
+The `convert` command from `ralph` allows to convert logs from edX to xAPI format.
+
+Run the following command:
+
+```
+# fetch edX archive, extract its content and convert them to xAPI format
+
+$ ralph fetch -b swift edx_archive.json | ralph extract -p gelf | ralph convert -f edx -t xapi -p
+```
+
+It returns directly in the terminal the whole bunch of converted logs.
+
+## Validate the conversion
+
+Then, you will pipe the output of the latter command set to `validate` that the conversion was successful.
+
+Run the following command:
+
+```
+# fetch edX archive, extract its content, convert them to xAPI format and validate the conversion
+
+$ ralph fetch -b swift edx_archive.json | ralph extract -p gelf | ralph convert -f edx -t xapi -p <...> | ralph validate -f xapi
+```
+
+## Push the converted logs to elasticsearch backend
+
+This tutorial will soon come to an end!
+
+If you went that far, it means that you have:
+
+1. pushed and fetched archives from Swift
+2. Then extract logs from the archives, convert them and validate the conversion to a new format
+
+Yet, you have to finally pushed the successfully converted data to the new backend, Elasticsearch.
+
+With the complete following command, you will push directly from Swift
+to elasticsearch by converting your logs "on-the-fly".
```

Comment:
```suggestion
to elasticsearch by converting your logs "on-the-fly".
```
Really not sure how to translate "à la volée" :grin:

Reply:
That's correct. :1st_place_medal: